1_fetch/src/summarize_wqp_records.R

```diff
@@ -0,0 +1,72 @@
+#' Function to summarize the expected number of sites and samples to be returned
+#' from the WQP for the area of interest
+#'
+#' @param wqp_inventory data frame containing sites returned from the WQP query,
+#' with one row per site; contains columns CharacteristicName and resultCount.
+#' @param fileout character string indicating the name of the saved file, including
+#' file path and extension.
+#' 
+summarize_wqp_inventory <- function(wqp_inventory, fileout){
+  
+  # Summarize WQP inventory
+  wqp_summary <- wqp_inventory %>%
+    group_by(CharacteristicName) %>%
+    summarize(sites_count = n(),
+              results_count = sum(resultCount),
+              .groups = 'drop') %>%
+    select(CharacteristicName, sites_count, results_count)
+  
+  # Write summary file
+  write_csv(wqp_summary, file = fileout)
+  
+  return(fileout)
+  
+}
+
+
+#' Summarize data downloaded from the Water Quality Portal
+#' 
+#' @description Function to summarize the data downloaded from the WQP and 
+#' compare with the number of sites and samples that were expected based on 
+#' the WQP inventory.
+#' 
+#' @param wqp_query_summary_csv character string indicating the name of the saved 
+#' query summary file, including file path and extension.
+#' @param wqp_data data frame containing the output from fetch_wqp_data; contains 
+#' data downloaded from the WQP, where each row represents a unique data record.
+#' @param fileout character string indicating the name of the saved file, including
+#' file path and extension.
+#'  
+summarize_wqp_data <- function(wqp_query_summary_csv, wqp_data, fileout){
+  
+  # Read in WQP query summary
+  expected_results <- readr::read_csv(wqp_query_summary_csv, 
+                                      show_col_types = FALSE) %>%
+    pull(results_count) %>%
+    sum()
+  realized_results <- dim(wqp_data)[1]
+  
+  # Compare WQP query summary with data pull 
+  if(realized_results != expected_results){
+    warning(sprintf("Was expecting %s rows of data based on the WQP inventory but data pull returned %s rows.", 
+                    expected_results, realized_results))
```

Comment:
I like this idea of comparing the two and communicating with the user! I am trying to think what a user might do if this happens? Could it be that the inventory was run earlier and now there are new results not captured by it? We should probably give the user some ideas of what to do to either reconcile or decide that it's OK and move on with no action.

Reply:
I wonder if more detail would be better? Such as, saying that certain characteristics are mismatched.

Reply:
I don't think it is too much text. I think it is much more helpful as a user because I could take action based on this message! When might the inventory be outdated but not built as part of a `tar_make()`? I would be more concerned about that potential pattern. Unless someone called `tar_make(p1_wqp_data_aoi, shortcut = TRUE)`, I think that `p1_wqp_inventory` would rebuild automatically if it was outdated.